src/components/views/dialogs/RoomSettingsDialog.tsx

```diff
@@ -132,6 +134,16 @@ class RoomSettingsDialog extends React.Component<IProps, IState> {
                 "RoomSettingsGeneral",
             ),
         );
+        if (SettingsStore.getValue("feature_ask_to_join") && this.state.room.getJoinRule() === "knock") {
```

Comment:
The tab doesn't appear/disappear correctly when the push rule is changed within the lifetime of this dialog being open

Reply:
Nice catch!